Sources/NIOCore/ByteBuffer-aux.swift

```diff
@@ -58,6 +58,50 @@ extension ByteBuffer {
         return result
     }
 
+    #if compiler(>=6.2)
+    @inlinable
+    @available(macOS 26, iOS 26, tvOS 26, watchOS 26, visionOS 26, *)
+    public mutating func readInlineArray<
+        let count: Int,
+        IntegerType: FixedWidthInteger
+    >(
+        endianness: Endianness = .big,
+        as: InlineArray<count, IntegerType>.Type = InlineArray<count, IntegerType>.self
+    ) -> InlineArray<count, IntegerType>? {
+        // use stride to account for padding bytes
+        let stride = MemoryLayout<IntegerType>.stride
+        let bytesRequired = stride * count
+
+        guard self.readableBytes >= bytesRequired else {
+            return nil
+        }
+
+        do {
+            let inlineArray = try InlineArray<count, IntegerType> { (outputSpan: inout OutputSpan<IntegerType>) in
+                for index in 0..<count {
+                    guard
+                        let integer = self.getInteger(
+                            // this is less than 'bytesRequired' so is safe to multiply
+                            at: stride &* index,
+                            endianness: endianness,
+                            as: IntegerType.self
+                        )
+                    else {
+                        throw InlineArrayFailedToGetElementError()
```

Comment:
This error is unreachable: if `readableBytes >= bytesRequired`, then there can be no way for this to throw, so we can force unwrap the return value of `getInteger`. Please do leave a comment in the code that reflects this fact.